### PR TITLE
Handle debugReports field in response

### PR DESCRIPTION
**Pull request by qingxinwu**
... to support downsampling forDebuggingOnly reports. And send debug report's lockout status in the request as part of enableDebugReports field.

### CODE REVIEW

File: draft-ietf-bidding-and-auction-services.md [hunk 1]
```diff
@@ -978,15 +992,40 @@ response from Bidding and Auction Services. It takes as input the
...
+      1. Let `ad tech origin` be `per origin debug reports["adTechOrigin"]` parsed as an [ORIGIN],
+         continue with the next iteration if there is an error.
+      1. If `per origin debug reports["reports"]` does not exist or is not an array,
+         continue with the next iteration.
+      1. For each `report` in `per origin debug reports["reports"]`:
+         1. If `report` is not a map, continue with the next iteration.
+         1. Let `component win` be `report["componentWin"]` if it exists and is a bool,
+            otherwise false.
+         1. If `report["url"]` exists and is a string:
+            1. Let `url` be `report["url"]` parsed as a [URL], or continue with the next
```
Comment (qingxinwu) [inline draft-ietf-bidding-and-auction-services.md:1013]: should we do https check in the IETF spec?

Comment (brusshamilton) [reply draft-ietf-bidding-and-auction-services.md]: I think it makes more sense to check for a "possibly trustworthy" URL in the W3C spec.

Comment (qingxinwu) [reply draft-ietf-bidding-and-auction-services.md]: ack. Thanks

File: draft-ietf-bidding-and-auction-services.md [hunk 2]
```diff
@@ -778,12 +779,8 @@ response = {
   ; If not present, map as Null.
   ? buyerAndSellerReportingId: tstr,
 
-  ; Optional SelectedBuyerAndSellerReportingId of the winning Ad
```
Comment (brusshamilton) [inline draft-ietf-bidding-and-auction-services.md:781]: We probably shouldn't be removing this, right?

File: draft-ietf-bidding-and-auction-services.md [hunk 3]
```diff
@@ -978,15 +992,48 @@ response from Bidding and Auction Services. It takes as input the
      parsed as a [URL], returning failure if there is an error.
 1. If `response["adMetadata"]` exists and is a string set
    `processed response["ad metadata"]` to `response["adMetadata"]`.
-1. If `response["buyerReportingId"]` exists and is a string, set
```
Comment (brusshamilton) [inline draft-ietf-bidding-and-auction-services.md:981]: We probably should keep the comma and selectedBuyerAndSellerReportingId parts.

Comment (qingxinwu) [reply draft-ietf-bidding-and-auction-services.md]: added back. Weird

**Review by brusshamilton — APPROVED** [review]

**Pull request merged**